Source/WebCore/rendering/svg/RenderSVGText.cpp

```diff
@@ -109,34 +109,30 @@ static inline void collectLayoutAttributes(RenderObject* text, Vector<SVGTextLay
     }
 }
 
-static inline bool findPreviousAndNextAttributes(RenderElement& start, RenderSVGInlineText* locateElement, bool& stopAfterNext, SVGTextLayoutAttributes*& previous, SVGTextLayoutAttributes*& next)
+static inline bool findPreviousAndNextAttributes(RenderElement& root, RenderSVGInlineText* locateElement, SVGTextLayoutAttributes*& previous, SVGTextLayoutAttributes*& next)
 {
     ASSERT(locateElement);
-    // FIXME: Make this iterative.
-    for (CheckedRef child : childrenOfType<RenderObject>(start)) {
-        if (auto* text = dynamicDowncast<RenderSVGInlineText>(child.get())) {
+    bool stopAfterNext = false;
+    CheckedPtr current = root.firstChild();
+    while (current) {
+        if (auto* text = dynamicDowncast<RenderSVGInlineText>(*current)) {
             if (locateElement != text) {
                 if (stopAfterNext) {
                     next = text->layoutAttributes();
                     return true;
                 }
-
                 previous = text->layoutAttributes();
+            } else
+                stopAfterNext = true;
+        } else if (auto* childSVGInline = dynamicDowncast<RenderSVGInline>(*current)) {
```

Comment:
I think you do not need this `dynamicDowncast` to access the `firstChild()` method. I think this should be enough
```
if (is<RenderSVGInline>(*current))
```